ibm/service/cis/resource_ibm_cis_alert_webhooks.go

```diff
@@ -0,0 +1,175 @@
+// Licensed under the Mozilla Public License v2.0
+
+package cis
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	cisWebhooksID     = "webhooks_id"
+	cisWebhooksName   = "name"
+	cisWebhooksURL    = "url"
+	cisWebhooksType   = "type"
+	cisWebhooksSecret = "secret"
+)
+
+func ResourceIBMCISWebhooks() *schema.Resource {
+	return &schema.Resource{
+		Create:   ResourceIBMCISWebhooksCreate,
+		Read:     ResourceIBMCISWebhooksRead,
+		Update:   ResourceIBMCISWebhooksUpdate,
+		Delete:   ResourceIBMCISWebhooksDelete,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisWebhooksID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Webhook ID",
+			},
+			cisWebhooksName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Webhook Name",
+			},
+			cisWebhooksURL: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Webhook URL",
+			},
+			cisWebhooksType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Webhook Type",
+			},
+			cisWebhooksSecret: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "API key needed to use the webhook",
+			},
+		},
+	}
+}
+func ResourceIBMCISWebhooksCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisWebhooksSession()
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while getting the CisWebhooksSession %s", err)
+	}
+
+	crn := d.Get(cisID).(string)
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewCreateAlertWebhookOptions()
+
+	if name, ok := d.GetOk(cisWebhooksName); ok {
+		opt.SetName(name.(string))
+	}
+	if url, ok := d.GetOk(cisWebhooksURL); ok {
+		opt.SetURL(url.(string))
+	}
+	if secret, ok := d.GetOk(cisWebhooksSecret); ok {
+		opt.SetSecret((secret.(string)))
+	}
+
+	result, resp, err := sess.CreateAlertWebhook(opt)
+	if err != nil || result == nil {
+		return fmt.Errorf("[ERROR] Error creating Webhooks  %s %s", err, resp)
+	}
+	log.Printf("Alert Webhooks created successfully : %s", *result.Result.ID)
+	d.SetId(flex.ConvertCisToTfTwoVar(*result.Result.ID, crn))
+	return ResourceIBMCISWebhooksRead(d, meta)
+
+}
+func ResourceIBMCISWebhooksRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisWebhooksSession()
+	if err != nil {
+		return err
+	}
+	webhooksID, crn, err := flex.ConvertTftoCisTwoVar(d.Id())
+	if err != nil {
+		return err
+	}
+	sess.Crn = core.StringPtr(crn)
+	opt := sess.NewGetWebhookOptions(webhooksID)
+
+	result, resp, err := sess.GetWebhook(opt)
+	if err != nil {
```

Comment:
`if err != nil || result == nil`

also check for 404.. if rc= 404 set id to "" and return nil